scripts/nlp_language_modeling/prepare_packed_ft_dataset.py

```diff
@@ -109,8 +123,26 @@ def tokenize_dataset(cfg: 'DictConfig'):
         special_tokens=data_cfg.get('chat_prompt_tokens', None),
         is_test=True,
     )
-
-    return np.array([dataset[i] for i in range(len(dataset))])
+    max_seq_length = dataset.max_seq_length
+    pad_id = dataset.tokenizer.eos_id
+    pad_seq_length_to_mult = dataset.pad_seq_length_to_mult
+    dataset = np.array([dataset[i] for i in range(len(dataset))])
+    if cp_size > 1:
+      def pre_pad_dataset(data, max_length, pad_id):
+        '''
+        pad each individual data point to the length of max_length
+        '''
+        for key,val in data.items():
+          if key in {'input_ids', 'context_ids'}:
+            val = val + [pad_id] * (max_length - len(val) + 1) # add 1 for cp
```

Comment:
Why add 1 for CP?

Reply:
I've documented the reason as a comment in the code: because input_ids is truncated by 1 for labels in the collate_fn of GPTSFTPackedDataset in gpt_sft_dataset.py, we add 1 extra padding here